@vates/node-vsphere-soap/lib/client.js

```diff
@@ -15,224 +15,218 @@
 const EventEmitter = require('events').EventEmitter
 const util = require('util')
 const soap = require('soap')
-const Cookie = require('soap-cookie')   // required for session persistence
-const { SSL_OP_NO_TLSv1_2 } = require('crypto').constants
-
+const Cookie = require('soap-cookie') // required for session persistence
+const https = require('https')
+const axios = require('axios')
 
 // Client class
 // inherits from EventEmitter
 // possible events: connect, error, ready
 
-function Client( vCenterHostname, username, password, sslVerify) {
-
-  this.status = 'disconnected';
-  this.reconnectCount = 0;
+function Client(vCenterHostname, username, password, sslVerify) {
+  this.status = 'disconnected'
+  this.reconnectCount = 0
 
-  sslVerify = typeof sslVerify !=='undefined' ? sslVerify : false;
+  sslVerify = typeof sslVerify !== 'undefined' ? sslVerify : false
 
-  EventEmitter.call( this );
+  EventEmitter.call(this)
 
   // sslVerify argument handling
-  if(sslVerify) {
-    this.clientopts = {};
+  if (sslVerify) {
+    this.clientopts = {}
   } else {
     this.clientopts = {
-      rejectUnauthorized: false,
-      strictSSL: false,
-      secureOptions: SSL_OP_NO_TLSv1_2 // likely needed for node >= 10.0
-    }; // recommended options by node-soap authors
-    process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0'; // need for self-signed certs
+      request: axios.create({
+        httpsAgent: new https.Agent({
+          rejectUnauthorized: false,
+        }),
+      }),
```

Comment:
the real change is here, we instantiate a request handler with the right options